lib/emscripten/src/env/windows/mod.rs

```diff
@@ -0,0 +1,225 @@
+/// NOTE: These syscalls only support wasm_32 for now because they take u32 offset
+use libc::{
+    c_int,
+    c_long,
+    getenv,
+    //sysconf, unsetenv,
+};
+
+use core::slice;
+use std::ffi::{CStr, CString};
+use std::mem;
+use std::os::raw::c_char;
+
+use crate::utils::{allocate_on_stack, copy_cstr_into_wasm, copy_terminated_array_of_cstrs};
+use crate::EmscriptenData;
+use wasmer_runtime_core::memory::Memory;
+use wasmer_runtime_core::vm::Ctx;
+
+#[link(name = "c")]
+extern "C" {
+    #[link_name = "_putenv"]
+    pub fn putenv(s: *const c_char) -> c_int;
+}
+
+pub fn _getaddrinfo(_one: i32, _two: i32, _three: i32, _four: i32, _ctx: &mut Ctx) -> i32 {
+    debug!("emscripten::_getaddrinfo");
+    -1
+}
+
+// #[no_mangle]
+/// emscripten: _getenv // (name: *const char) -> *const c_char;
+pub fn _getenv(name: i32, ctx: &mut Ctx) -> u32 {
+    debug!("emscripten::_getenv");
+
+    let memory = ctx.memory(0);
+
+    let name_addr = emscripten_memory_pointer!(ctx.memory(0), name) as _;
+
+    debug!("=> name({:?})", unsafe { CStr::from_ptr(name_addr) });
+
+    let c_str: *mut c_char = unsafe { getenv(name_addr as _) } as _;
+    if c_str.is_null() {
+        return 0;
+    }
+
+    unsafe { copy_cstr_into_wasm(ctx, c_str) }
+}
+
+unsafe fn read_string_from_wasm(pointer: c_int, memory: &Memory) -> String {
+    let string_cell = memory.view::<u8>()[(pointer as usize)..].as_ptr();
+    let string_ptr = string_cell.as_ptr();
+    let cstr_ptr = CStr::from_ptr(string_ptr);
+    String::From(cstr_ptr)
+}
+
+/// emscripten: _setenv // (name: *const char, name: *const value, overwrite: int);
+pub fn _setenv(name: c_int, value: c_int, overwrite: c_int, ctx: &mut Ctx) -> c_int {
+    debug!("emscripten::_setenv");
+    let name_addr = emscripten_memory_pointer!(ctx.memory(0), name);
+    let value_addr = emscripten_memory_pointer!(ctx.memory(0), value);
+    // setenv does not exist on windows, so we hack it with _putenv
+    unsafe {
+        let name = read_string_from_wasm(name, ctx.memory(0));
+        let value = read_string_from_wasm(value, ctx.memory(0));
+        let putenv_string = format!("{}={}", name, value);
+        let putenv_cstring = CString::from::<String>(putenv_string);
+        let putenv_raw_ptr = putenv_cstring.as_ptr();
+        debug!("=> name({:?})", CStr::from_ptr(name_addr));
+        debug!("=> value({:?})", CStr::from_ptr(value_addr));
+        putenv(putenv_raw_ptr)
+    }
+}
```

Comment:
this was re-written and uses a helper function to pull out the strings and concatenate them

Reply:
Try to reduce the unsafe block to just the last line.